### PR TITLE
Add success and fail redirects to email login link

### PR DESCRIPTION
**Pull request by tspenov**
## Changes

```graphql
mutation {
  emailLogin(email: "john@example.com", failRedirectUrl: "https://app.santiment.net/fail", successRedirectUrl: "https://app.santiment.net/success") {
    success
    firstLogin
  }
}
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/accounts/user/user_email.ex [hunk 1]
```diff
@@ -154,6 +154,16 @@ defmodule Sanbase.Accounts.User.Email do
...
+    query_map =
+      if valid_redirect_url?(args[:success_redirect_url]),
+        do: Map.put(query_map, :success_redirect_url, args[:success_redirect_url]),
+        else: query_map
+
+    query_map =
+      if valid_redirect_url?(args[:fail_redirect_url]),
+        do: Map.put(query_map, :fail_redirect_url, args[:fail_redirect_url]),
+        else: query_map
+
```
Comment (IvanIvanoff) [inline lib/sanbase/accounts/user/user_email.ex:166]: Hm, what if you do in a `with` statement?
```elixir
with {:ok, query_map} <- maybe_add_url(query_map, args, :success_redirect_url)
         {:ok, query_map} <- maybe_add_url(query_map, args, :fail_redirect_url) dO
         ...
```
so we don't hide errors for invalid URLs?

File: lib/sanbase/accounts/user/user_email.ex [hunk 2]
```diff
@@ -164,4 +174,17 @@ defmodule Sanbase.Accounts.User.Email do
...
+    case URI.parse(url).host do
+      nil -> false
+      host -> allowed_url?(String.split(host, "."))
+    end
+  end
+
+  def valid_redirect_url?(_), do: false
+
+  defp allowed_url?(["santiment", "net"]), do: true
+  defp allowed_url?([_app, "santiment", "net"]), do: true
```
Comment (IvanIvanoff) [inline lib/sanbase/accounts/user/user_email.ex:188]: nitpick: the subdomain can be `insights`, `queries`, etc., so this `_app` is a little bit misleading.

**Review by IvanIvanoff — APPROVED** [review]

File: lib/sanbase/accounts/user/user_email.ex (hunk 2 above)
Comment (tspenov) [reply lib/sanbase/accounts/user/user_email.ex:188]: it was meant not for `app` but for application

File: lib/sanbase/accounts/user/user_email.ex (hunk 1 above)
Comment (tspenov) [reply lib/sanbase/accounts/user/user_email.ex:166]: you mean to return the error to the frontend if of one of redirect urls is malformed ?

**Pull request merged**